File: Python/LastStoneWeight.py
import heapq

def lastStoneWeight(stones):

    stones = [-s for s in stones]
    heapq.heapify(stones)

    while len(stones) > 1:
        first = heapq.heappop(stones)
        second = heapq.heappop(stones)
        if second > first:
            heapq.push(stones,first-second)

    stones.append(0)
    return -1*stones[0]

File: Javascript/LastStoneWeight.js
import { MaxPriorityQueue} from "datastructures-js/priority-queue"

var lastStoneWeight = function(stones){

    const maxHeap = new MaxPriorityQueue()
    for(const stone of stones){
        maxHeap.enqueue(stone)
    }

    while(maxHeap.size() > 1){
        const [first,second] = [maxHeap.dequeue().element,maxHeap.dequeue().element]
        if (first > second){
            maxHeap.enqueue(first-second)
        }
    }

    maxHeap.enqueue(0)
    return maxHeap.dequeue().element
}